python/pyspark/rdd.py

```diff
@@ -2587,6 +2616,7 @@ def pipeline_func(split, iterator):
             self._prev_jrdd = prev._prev_jrdd  # maintain the pipeline
             self._prev_jrdd_deserializer = prev._prev_jrdd_deserializer
         self.is_cached = False
+        self.has_resourceProfile = False
```

Comment:
nit `has_resourceProfile` -> `has_resource_profile`. Basically the logic is that we should keep everything `a_b_c` when it's not an API inspired from Java (or Scala API). IIRC, this logic complies pep8.